mobile/src/full/java/org/openhab/habdroid/core/GcmIntentService.java

```diff
@@ -64,7 +66,19 @@ protected void onHandleIntent(Intent intent) {
                     notificationId = Integer.parseInt(intent.getExtras().getString(EXTRA_NOTIFICATION_ID));
                 }
                 if ("notification".equals(intent.getExtras().getString("type"))) {
-                    sendNotification(intent.getExtras().getString(EXTRA_MSG), notificationId);
+                    //we use local time as fallback
+                    long timestamp = System.currentTimeMillis();
+                    if(intent.getExtras().containsKey(EXTRA_TIMESTAMP)){
```

Comment:
Are there any plans on openhab-cloud to support this?

Reply:
I created an issue in the cloud repository at least: https://github.com/openhab/openhab-cloud/issues/160

Reply:
Should we merge this change before the feature exists server side? I dont think so.  We can wait until its in openhab-cloud or just merge the time from gcm for now and timestamp later.